In [1]:
# Import Dependencies
import pandas as pd

In [2]:
# File Paths

file_path_one = ('../../data/data.csv')
file_path_two = ('../../data/heart_statlog_cleveland_hungary_final.csv')

In [3]:
# Read the 1st csv - smaller dataset with 2 extra features
hd_small_df = pd.read_csv(file_path_one)
hd_small_df.head()

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  slope  \
0   70    1   4       130   322    0        2      109      0      2.4      2   
1   67    0   3       115   564    0        2      160      0      1.6      2   
2   57    1   2       124   261    0        0      141      0      0.3      1   
3   64    1   4       128   263    0        0      105      1      0.2      2   
4   74    0   2       120   269    0        2      121      1      0.2      1   

   ca  thal  target  
0   3     3       2  
1   0     7       1  
2   0     7       2  
3   1     7       1  
4   1     3       1

In [4]:
# Read the 2nd csv - larger dataset with less features
hd_large_df = pd.read_csv(file_path_two)
hd_large_df.head()

age  sex  chest pain type  resting bp s  cholesterol  fasting blood sugar  \
0   40    1                2           140          289                    0   
1   49    0                3           160          180                    0   
2   37    1                2           130          283                    0   
3   48    0                4           138          214                    0   
4   54    1                3           150          195                    0   

   resting ecg  max heart rate  exercise angina  oldpeak  ST slope  target  
0            0             172                0      0.0         1       0  
1            0             156                0      1.0         2       1  
2            1              98                0      0.0         1       0  
3            0             108                1      1.5         2       1  
4            0             122                0      0.0         1       0

age: age in years

sex: sex (1 = male; 0 = female)

cp: chest pain type (Value 0: typical angina; Value 1: atypical angina; Value 2: non-anginal pain; Value 3: asymptomatic)

trestbps: resting blood pressure in mm Hg on admission to the hospital

chol: serum cholestoral in mg/dl

fbs: fasting blood sugar > 120 mg/dl (1 = true; 0 = false)

restecg: resting electrocardiographic results (Value 0: normal; Value 1: having ST-T wave abnormality; Value 2: 
probable or definite left ventricular hypertrophy)

thalach: maximum heart rate achieved

exang: exercise induced angina (1 = yes; 0 = no)

oldpeak: ST depression induced by exercise relative to rest

slope: the slope of the peak exercise ST segment (Value 0: upsloping; Value 1: flat; Value 2: downsloping)

ca: number of major vessels (0-3) colored by flourosopy

thal: thalassemia (3 = normal; 6 = fixed defect; 7 = reversable defect)

target: heart disease (1 = no, 2 = yes)

In [5]:
# Dropping na on Small dataset
hd_small_df.dropna(inplace = True)

In [6]:
# Renaming columns in the Small Dataset
hd_small_df.rename(columns={'cp':'chest pain type', 
                            'trestbps': 'resting bp s', 
                            'chol':'cholesterol',
                            'fbs':'fasting blood sugar',
                            'restecg':'resting ecg',
                            'thalach':'max heart rate',
                            'exang':'exercise angina',
                            'slope':'ST slope',
                            'ca':'major vessels',
                            'thal':'thalassemia'}, inplace = True)
hd_small_df

age  sex  chest pain type  resting bp s  cholesterol  \
0     70    1                4           130          322   
1     67    0                3           115          564   
2     57    1                2           124          261   
3     64    1                4           128          263   
4     74    0                2           120          269   
..   ...  ...              ...           ...          ...   
265   52    1                3           172          199   
266   44    1                2           120          263   
267   56    0                2           140          294   
268   57    1                4           140          192   
269   67    1                4           160          286   

     fasting blood sugar  resting ecg  max heart rate  exercise angina  \
0                      0            2             109                0   
1                      0            2             160                0   
2                      0            0             141                0   
3                      0            0             105                1   
4                      0            2             121                1   
..                   ...          ...             ...              ...   
265                    1            0             162                0   
266                    0            0             173                0   
267                    0            2             153                0   
268                    0            0             148                0   
269                    0            2             108                1   

     oldpeak  ST slope  major vessels  thalassemia  target  
0        2.4         2              3            3       2  
1        1.6         2              0            7       1  
2        0.3         1              0            7       2  
3        0.2         2              1            7       1  
4        0.2         1              1            3       1  
..       ...       ...            ...          ...     ...  
265      0.5         1              0            7       1  
266      0.0         1              0            7       1  
267      1.3         2              0            3       1  
268      0.4         2              0            6       1  
269      1.5         2              3            3       2  

[270 rows x 14 columns]

In [8]:
# Changing Target values from 1 to 0 and 2 to 1 to be read easier for ML

hd_small_df["target"].replace({1: 0, 2: 1}, inplace=True)
hd_small_df.head()

age  sex  chest pain type  resting bp s  cholesterol  fasting blood sugar  \
0   70    1                4           130          322                    0   
1   67    0                3           115          564                    0   
2   57    1                2           124          261                    0   
3   64    1                4           128          263                    0   
4   74    0                2           120          269                    0   

   resting ecg  max heart rate  exercise angina  oldpeak  ST slope  \
0            2             109                0      2.4         2   
1            2             160                0      1.6         2   
2            0             141                0      0.3         1   
3            0             105                1      0.2         2   
4            2             121                1      0.2         1   

   major vessels  thalassemia  target  
0              3            3       1  
1              0            7       0  
2              0            7       1  
3              1            7       0  
4              1            3       0

In [9]:
# Dropping na on Large dataset
hd_large_df.dropna(inplace = True)
hd_large_df

age  sex  chest pain type  resting bp s  cholesterol  \
0      40    1                2           140          289   
1      49    0                3           160          180   
2      37    1                2           130          283   
3      48    0                4           138          214   
4      54    1                3           150          195   
...   ...  ...              ...           ...          ...   
1185   45    1                1           110          264   
1186   68    1                4           144          193   
1187   57    1                4           130          131   
1188   57    0                2           130          236   
1189   38    1                3           138          175   

      fasting blood sugar  resting ecg  max heart rate  exercise angina  \
0                       0            0             172                0   
1                       0            0             156                0   
2                       0            1              98                0   
3                       0            0             108                1   
4                       0            0             122                0   
...                   ...          ...             ...              ...   
1185                    0            0             132                0   
1186                    1            0             141                0   
1187                    0            0             115                1   
1188                    0            2             174                0   
1189                    0            0             173                0   

      oldpeak  ST slope  target  
0         0.0         1       0  
1         1.0         2       1  
2         0.0         1       0  
3         1.5         2       1  
4         0.0         1       0  
...       ...       ...     ...  
1185      1.2         2       1  
1186      3.4         2       1  
1187      1.2         2       1  
1188      0.0         2       1  
1189      0.0         1       0  

[1190 rows x 12 columns]

In [10]:
# Loading the cleaned dataframes to new CSV files
hd_small_df.to_csv('../../data/small_heart_disease_dataset.csv', index = False)
hd_large_df.to_csv('../../data/large_heart_disease_dataset.csv', index = False)